TST: instability with scipy 1.13.0 on macOS arm64 (floating point comparison)

With scipy 1.13.0 on macOS arm64, I'm seeing the following failure
```
pytest docs/coordinates/matchsep.rst
```

<details><summary>Pytest log</summary>

```
===================== test session starts ======================
platform darwin -- Python 3.12.2, pytest-8.1.1, pluggy-1.5.0

Running tests with Astropy version 7.0.0.dev41+gdcf60561fc.
Running tests in astropy docs.

Date: 2024-04-22T10:33:40

Platform: macOS-14.4.1-arm64-arm-64bit

Executable: /Users/clm/.pyenv/versions/astropy.dev/bin/python

Full Python Version:
3.12.2 (main, Feb 22 2024, 09:53:39) [Clang 15.0.0 (clang-1500.1.0.2.5)]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions:
Numpy: 1.26.4
Scipy: 1.13.0
Matplotlib: not available
h5py: not available
Pandas: not available
PyERFA: 2.0.1.4
Cython: not available
Scikit-image: not available
asdf-astropy: not available
pyarrow: not available

Using Astropy options: remote_data: none.

CI: undefined
ARCH_ON_CI: undefined
IS_CRON: undefined

rootdir: /Users/clm/dev/astropy-project/coordinated/astropy
configfile: pyproject.toml
testpaths: astropy, docs
plugins: astropy-0.11.0, cov-5.0.0, hypothesis-6.100.1, remotedata-0.4.1, filter-subpackage-0.2.0, doctestplus-1.2.1, astropy-header-0.2.2, arraydiff-0.6.1, xdist-3.5.0, mock-3.14.0
collected 28624 items / 28623 deselected / 53 skipped / 1 selected
run-last-failure: rerun previous 1 failure (skipped 362 files)

docs/coordinates/matchsep.rst F                          [100%]

=========================== FAILURES ===========================
____________________ [doctest] matchsep.rst ____________________
244 Now ``idx`` are indices into ``catalog`` that are the closest objects to each
245 of the coordinates in ``c``, ``d2d`` are the on-sky distances between them, and
246 ``d3d`` are the 3-dimensional distances. Because coordinate objects support
247 indexing, ``idx`` enables easy access to the matched set of coordinates in
248 the catalog:
249
250 .. doctest-requires:: scipy
251
252     >>> matches = catalog[idx]
253     >>> (matches.separation_3d(c) == d3d).all()
Expected:
    True
Got:
    False

/Users/clm/dev/astropy-project/coordinated/astropy/docs/coordinates/matchsep.rst:253: DocTestFailure
=================== short test summary info ====================
FAILED docs/coordinates/matchsep.rst::matchsep.rst
======= 1 failed, 53 skipped, 28623 deselected in 7.43s ========
```

</details>

This is a floating point comparison between two arrays that appear to be at a machine-precision level difference from one another in scipy 1.13.0. Most likely, this isn't a bug in scipy, and the correct fix might simply be to adjust this doctest on our side to add a very small margin of error to account for floating point fuzziness.

--- a/docs/coordinates/matchsep.rst
+++ b/docs/coordinates/matchsep.rst
@@ -249,9 +249,11 @@ the catalog:
 
 .. doctest-requires:: scipy
 
+    >>> d3d # doctest: +FLOAT_CMP
+    <Quantity [1335.55538257] kpc>
     >>> matches = catalog[idx]
-    >>> (matches.separation_3d(c) == d3d).all()
-    True
+    >>> matches.separation_3d(c) # doctest: +FLOAT_CMP
+    <Distance [1335.55538257] kpc>
     >>> dra, ddec = c.spherical_offsets_to(matches)
 
 This functionality can also be accessed from the

